### PR TITLE
Sprint 11 (dev->stg)

### PR DESCRIPTION
**Pull request by burgwyn**
#214 
- Add DOB to user profile
- Add phone number to user profile
- Add migrations
- Allow DOB/Phone number writes by user

#202 
- Adds logic to bid view to prevent bids on positions that have other bids with handshakes (or above)
- Adds tests

#212 
- Add fairshare property to user
- Fairshare is calculated on-the-fly based on the users assignments using standard fairshare rules

### CODE REVIEW

File: talentmap_api/bidding/views/bidlist.py [hunk 1]
```diff
@@ -110,6 +110,14 @@ def put(self, request, pk, format=None):
         bidcycle = BidCycle.objects.filter(active=True).latest("cycle_start_date")
         position = get_object_or_404(bidcycle.positions.all(), id=pk)
 
+        # Position must not already have a handshake or greater bid status
+        bid_statuses = list(position.bids.filter(bidcycle=bidcycle).values_list("status", flat=True))
+        if (Bid.Status.handshake_offered in bid_statuses or
+                Bid.Status.handshake_accepted in bid_statuses or
+                Bid.Status.in_panel in bid_statuses or
+                Bid.Status.approved in bid_statuses):
+            return Response("Cannot bid on a position with a handshake", status=status.HTTP_400_BAD_REQUEST)
```
Comment (jseppi) [inline talentmap_api/bidding/views/bidlist.py]: The error message here isn't quite right now that we have `in_panel` and `approved` in the list of disqualifying statuses. Maybe something more generic like `"Cannot bid on a position that already has a qualifying bid"`.

File: talentmap_api/bidding/views/bidlist.py [hunk 2]
```diff
@@ -110,6 +110,14 @@ def put(self, request, pk, format=None):
         bidcycle = BidCycle.objects.filter(active=True).latest("cycle_start_date")
         position = get_object_or_404(bidcycle.positions.all(), id=pk)
 
+        # Position must not already have a handshake or greater bid status
+        bid_statuses = list(position.bids.filter(bidcycle=bidcycle).values_list("status", flat=True))
+        if (Bid.Status.handshake_offered in bid_statuses or
```
Comment (jseppi) [inline talentmap_api/bidding/views/bidlist.py]: hrm, could this logic maybe be encapsulated on the `Position` model into like a `can_accept_new_bids()` method? I don't know if that'll be better, but big `if` statements like this are both hard to read and easy to miss if logic needs to change.

File: talentmap_api/bidding/views/bidlist.py (hunk 1 above)
Comment (burgwyn) [reply talentmap_api/bidding/views/bidlist.py]: We discussed this and decided to hold off on getting language from the team.  We could do something simple like this though.

Comment (afrasier) [reply talentmap_api/bidding/views/bidlist.py]: Sounds good!

File: talentmap_api/bidding/views/bidlist.py (hunk 2 above)
Comment (afrasier) [reply talentmap_api/bidding/views/bidlist.py]: Makes sense, will do.

**Pull request merged**